app/src/main/java/com/github/se/wanderpals/ui/screens/overview/OverviewTrip.kt

```diff
@@ -68,21 +98,37 @@ fun OverviewTrip(trip: Trip, navigationActions: NavigationActions) {
                         letterSpacing = 0.5.sp,
                     ))
             Spacer(modifier = Modifier.height(3.dp))
-            // Start date
-            Text(
-                text =
-                    "From : %s"
-                        .format(trip.startDate.format(DateTimeFormatter.ofPattern(DATE_PATTERN))),
-                modifier = Modifier.height(24.dp),
-                style =
-                    TextStyle(
-                        fontSize = 12.sp,
-                        lineHeight = 24.sp,
-                        fontWeight = FontWeight(500),
-                        color = Color(0xFF000000),
-                        letterSpacing = 0.5.sp,
-                    ))
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween) {
+                  // Start date
+                  Text(
+                      text =
+                          "From : %s"
+                              .format(
+                                  trip.startDate.format(DateTimeFormatter.ofPattern(DATE_PATTERN))),
+                      modifier = Modifier.height(24.dp),
+                      style =
+                          TextStyle(
+                              fontSize = 12.sp,
+                              lineHeight = 24.sp,
+                              fontWeight = FontWeight(500),
+                              color = Color(0xFF000000),
+                              letterSpacing = 0.5.sp,
+                          ))
+                  // Share trip code button
+                  IconButton(
+                      modifier = Modifier.size(20.dp).testTag("shareTripButton" + trip.tripId),
+                      onClick = { context.shareTripCodeIntent(trip.tripId) },
```

Comment:
It's good, one thing that can be improved is: there's a background color (let's say grey) for the share icon when clicking on it so that the user will not be confused about which thing (the box button for going to the dashboard or the icon button for sharing the trip) he/she is clicking

Reply:
Nice catch ! I did not thought about this small detail.